packages/fgtclb/academic-jobs/Classes/Property/TypeConverter/ImageUploadConverter.php

```diff
@@ -100,36 +105,55 @@ public function convertFrom(
-            throw new TypeConverterException('Uploading files with PHP file extensions is not allowed!', 1690525745);
+    private function importUploadedResource(
+        UploadedFile $uploadedFileInformation,
+        string $targetFolderIdentifier,
+        ?string $targetFileName
+    ): ExtbaseFileReference {
+        if (!GeneralUtility::makeInstance(FileNameValidator::class)->isValid((string)$uploadedFileInformation->getClientFilename())) {
+            throw new TypeConverterException('Uploading files with PHP file extensions is not allowed!', 1753712929);
         }
 
         $targetFolder = $this->getOrCreateTargetFolder($targetFolderIdentifier);
-        $uploadedFile = $targetFolder->addUploadedFile($uploadedFileInformation, DuplicationBehavior::REPLACE);
+        /** @var File $uploadedFile */
+        $uploadedFile = $targetFolder->getStorage()->addUploadedFile(
+            $uploadedFileInformation,
+            $targetFolder,
+            $targetFileName,
+            DuplicationBehavior::REPLACE
+        );
 
-        $resourcePointer = isset($uploadedFileInformation['__identity']) ? (int)$uploadedFileInformation['__identity'] : null;
-        return $this->createFileReferenceFromFalFileObject($uploadedFile, $resourcePointer);
+        return $this->createFileReferenceFromFalFileObject($uploadedFile);
     }
 
     /**
-     * @param array{size: int, type: string} $uploadedFileInformation
      * @throws TypeConverterException
      */
-    private function validateUploadedFile(array $uploadedFileInformation, string $maxFileSize, string $allowedMimeTypes): void
+    private function validateUploadedFile(UploadedFile $uploadedFileInformation, string $maxFileSize, string $allowedMimeTypes): void
     {
+        $typoScriptFrontendController = $this->getTypo3Request()->getAttribute('frontend.controller') ?? $GLOBALS['TSFE'] ?? null;
         $maxFileSizeInBytes = GeneralUtility::getBytesFromSizeMeasurement($maxFileSize);
         $allowedMimeTypesArray = GeneralUtility::trimExplode(',', $allowedMimeTypes);
 
-        if ($uploadedFileInformation['size'] > $maxFileSizeInBytes) {
-            throw new TypeConverterException('Uploaded file exceeds allowed file size', 1690538138);
+        if ($uploadedFileInformation->getSize() > $maxFileSizeInBytes) {
+            throw new TypeConverterException(
+                $typoScriptFrontendController?->sL(
+                    'LLL:EXT:form/Resources/Private/Language/locallang.xlf:upload.error.150530345'
+                ) ?? 'Upload error',
+                1753712943
+            );
         }
-        if (!in_array($uploadedFileInformation['type'], $allowedMimeTypesArray, true)) {
-            throw new TypeConverterException('The uploaded file type is not allowed', 1721647425);
+        if (!in_array($uploadedFileInformation->getClientMediaType(), $allowedMimeTypesArray, true)) {
+            throw new TypeConverterException(
+                $typoScriptFrontendController?->sL(
```

Comment:
Why not getting the `LanguageServiceFactory` injected (constructor di) and creating the `LangaugeService` using `->createFromSiteLanguage()` with the site from the request and calling `sL` on that ? TSFE is scheduled to be removed in v14 (deprecated in v13) .... 

Cannot see other TSFE usage here .. so lets make the future prove directly.